src/super_gradients/training/datasets/data_formats/bbox_formats/cxcywh.py

```diff
@@ -94,7 +94,7 @@ def xyxy_to_cxcywh_inplace(bboxes, image_shape: Tuple[int, int]):
                 f"Detected non floating-point ({bboxes.dtype}) input to xyxy_to_cxcywh_inplace function. This may cause rounding errors and lose of precision. "
                 "You may want to convert your array to floating-point precision first."
             )
-        if isinstance(bboxes, np.ndarray) and not isinstance(bboxes.dtype, np.floating):
+        if isinstance(bboxes, np.ndarray) and not isinstance(bboxes.dtype.type, np.floating):
```

Comment:
Nice catch! If it isn't too much work, can you please help us improve this part:
1) Introduce `is_floating_point_array(x:np.ndarray)->bool` function that, by it's name would return True or False depending whether the input argument is an array of floating-point values
2) Replace usage here with this funciton
3) Add a unit test to test this function:
```
assertTrue(is_floating_point_array(np.zeros((32,32), dtype=np.float16))
assertTrue(is_floating_point_array(np.zeros((32,32), dtype=np.float32))
assertTrue(is_floating_point_array(np.zeros((32,32), dtype=np.float64))
assertFalse(is_floating_point_array(np.zeros((32,32), dtype=int))
assertFalse(is_floating_point_array(np.zeros((32,32), dtype=np.int32))
assertFalse(is_floating_point_array(np.zeros((32,32), dtype=np.bool))
```

Thanks in advance!

Reply:
A great idea, my initial impl eventually didn't pass all tests, the result looked pretty random (env-dependent). All good now 
4a9060c